invenio_circulation/api.py

```diff
@@ -92,6 +95,41 @@ def __init__(self, data, model=None):
             finalize_event='save',
         )
 
+    @classmethod
+    def item_available(cls, item_pid):
+        """."""
+        item_available = current_app.config.get('CIRCULATION_POLICIES').get(
+            'item_available'
+        )
+        if not item_available(item_pid):
+            return False
+        for loan in cls.get_loans(item_pid):
+            if loan.state != 'ITEM_RETURNED':
```

Comment:
Isn't it possible to ask elastic search to filter beforehand the the loans with status 'ITEM_RETURNED' ?

Reply:
Agree, why not having:
`config.py`
```
CIRCULATION_STATES_ITEM_AVAILABLE = ['PENDING', 'ITEM_RETURNED', ....]
```

and then in `api.py`
```
def is_item_available(item_pid):
    ... search in ES ...
    return "no loans with this item_pid" OR "item_pid=item_pid and state in config['CIRCULATION_STATES_ITEM_AVAILABLE']"

Reply:
Agree